library/src/main/java/com/android/volley/CacheDispatcher.java

```diff
@@ -134,10 +134,21 @@ public void run() {
                         new NetworkResponse(entry.data, entry.responseHeaders));
                 request.addMarker("cache-hit-parsed");
 
-
                 if (!entry.refreshNeeded()) {
-                    // Completely unexpired cache hit. Just deliver the response.
-                    mDelivery.postResponse(request, response);
+                    if (entry.softTtl - System.currentTimeMillis() < request.getSoftTTL() / 2) {
```

Comment:
make this a multiply by .5 and pull it out to a const at the top so we can tune if necessary